lifecycle/export.go

```diff
@@ -0,0 +1,149 @@
+package lifecycle
+
+import (
+	"encoding/json"
+	artUtils "github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	artServices "github.com/jfrog/jfrog-client-go/artifactory/services"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	clientConfig "github.com/jfrog/jfrog-client-go/config"
+	"github.com/jfrog/jfrog-client-go/lifecycle/services"
+	utils2 "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"strings"
+)
+
+type ReleaseBundleExportCommand struct {
+	releaseBundleCmd
+	modifications          *services.Modifications
+	downloadConfigurations *artUtils.DownloadConfiguration
+}
+
+func (rbe *ReleaseBundleExportCommand) Run() (err error) {
+	if err = validateArtifactoryVersionSupported(rbe.serverDetails); err != nil {
+		return
+	}
+	servicesManager, rbDetails, queryParams, err := rbe.getPrerequisites()
+	if err != nil {
+		log.Debug("Failed getting prerequisites for exporting command, error: ", err.Error())
+		return
+	}
+	// Start the Export process and wait for completion
+	log.Info("Exporting Release Bundle archive...")
+	releaseBundleExportParams := NewReleaseBundleExportParams(rbDetails, *rbe.modifications)
+	exportResponse, err := servicesManager.ExportReleaseBundle(releaseBundleExportParams, queryParams)
+	if err != nil {
+		log.Debug("Failed exporting release bundle, error: ", err.Error())
+		return
+	}
+	// Download the exported bundle
+	log.Debug("Downloading the exported bundle...")
+	cleanUp, err := rbe.downloadReleaseBundle(exportResponse, rbe.downloadConfigurations)
+	defer func() {
+		err = cleanUp()
+	}()
+	if err != nil {
+		return
+	}
+	log.Info("Successfully Downloaded Release Bundle archive")
+	content, err := json.Marshal(exportResponse)
+	if err != nil {
+		return err
+	}
+	log.Output(utils2.IndentJson(content))
+	return
+}
+
+// Download the exported release bundle using artifactory service manager
+func (rbe *ReleaseBundleExportCommand) downloadReleaseBundle(exportResponse services.ReleaseBundleExportedStatusResponse, downloadConfiguration *artUtils.DownloadConfiguration) (cleanUp func() error, err error) {
+	downloadParams := artServices.DownloadParams{
+		CommonParams: &utils.CommonParams{
+			Pattern: strings.TrimPrefix(exportResponse.RelativeUrl, "/"),
+		},
+		Symlink:         downloadConfiguration.Symlink,
+		ValidateSymlink: downloadConfiguration.ValidateSymlink,
+		MinSplitSize:    downloadConfiguration.MinSplitSize,
+		SplitCount:      downloadConfiguration.SplitCount,
+		SkipChecksum:    downloadConfiguration.SkipChecksum,
+	}
+	artifactoryServiceManager, err := createArtifactoryServiceManager(rbe.serverDetails)
```

Comment:
In order for the progress bar to work, we need to initialize a service manager with progress (see `CreateDownloadServiceManager` for example). This function doesn't do it.
Let's be sure we want to add this at this stage of time. If we do, it will require a bit more configuration both here and on the CLI side.

Reply:
As for this, I've tested with a 1GB bundle at the logs seems pretty well.
I guess if we will see the need in the future we can implement a full progress bar.